src/h3lib/lib/vertex.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file  vertex.h
+ *  @brief Functions for working with cell vertexes.
+ */
+
+#include "vertex.h"
+
+#include "baseCells.h"
+#include "faceijk.h"
+#include "geoCoord.h"
+#include "h3Index.h"
+
+/** @brief Table of direction-to-face mapping for each pentagon
+ *
+ * Note that faces are in directional order, starting at J_AXES_DIGIT.
+ * This table is generated by the generatePentagonDirectionFaces script.
+ */
+static const PentagonDirectionFaces pentagonDirectionFaces[NUM_PENTAGONS] = {
+    {4, {4, 0, 2, 1, 3}},       {14, {6, 11, 2, 7, 1}},
+    {24, {5, 10, 1, 6, 0}},     {38, {7, 12, 3, 8, 2}},
+    {49, {9, 14, 0, 5, 4}},     {58, {8, 13, 4, 9, 3}},
+    {63, {11, 6, 15, 10, 16}},  {72, {12, 7, 16, 11, 17}},
+    {83, {10, 5, 19, 14, 15}},  {97, {13, 8, 17, 12, 18}},
+    {107, {14, 9, 18, 13, 19}}, {117, {15, 19, 17, 18, 16}},
+};
+
+/**
+ * Get the number of CCW rotations of the cell's vertex numbers
+ * compared to the directional layout of its neighbors.
+ * @return Number of CCW rotations for the cell
+ */
+int vertexRotations(H3Index cell) {
+    // Get the face and other info for the origin
+    FaceIJK fijk;
+    _h3ToFaceIjk(cell, &fijk);
+    int baseCell = H3_EXPORT(h3GetBaseCell)(cell);
+    int cellLeadingDigit = _h3LeadingNonZeroDigit(cell);
+
+    // get the base cell face
+    FaceIJK baseFijk;
+    _baseCellToFaceIjk(baseCell, &baseFijk);
+
+    int ccwRot60 = _baseCellToCCWrot60(baseCell, fijk.face);
+
+    if (_isBaseCellPentagon(baseCell)) {
+        // Find the appropriate direction-to-face mapping
+        PentagonDirectionFaces dirFaces;
+        for (int p = 0; p < NUM_PENTAGONS; p++) {
+            if (pentagonDirectionFaces[p].baseCell == baseCell) {
+                dirFaces = pentagonDirectionFaces[p];
+            }
+        }
+
+        // additional CCW rotation for polar neighbors or IK neighbors
+        if (fijk.face != baseFijk.face &&
+            (_isBaseCellPolarPentagon(baseCell) ||
+             fijk.face == dirFaces.faces[IK_AXES_DIGIT - 2])) {
```

Comment:
I find this table a little confusing to use because it starts indexing directions starting with J (2) rather than 0 or 1. Perhaps at least extract the `2` to a reused constant for this function?

Reply:
Will extract to a const. The alternative was adding two `INVALID_FACE` entries to each array in the table, which didn't seem worthwhile.

Reply:
Added as a defined macro (I put this in the `.c` file itself, rather than the header file, because this seemed clearer and it didn't make sense to export it - let me know if you disagree).